packages/flutter_tools/lib/src/version.dart

```diff
@@ -405,6 +408,85 @@ class FlutterVersion {
   }
 }
 
+/// Checks if the provided [version] is tracking a standard remote.
+///
+/// A "standard remote" is one having the same url as [globals.flutterGit].
+///
+/// To initiate the validation check, call [run].
+///
+/// This prevents the tool from checking for version freshness in
+/// [globals.flutterGit], and fetching updates from upstream of the current
+/// branch/channel, which can be different.
+///
+/// This also prevents unnecessary freshness check from a forked repo unless the
+/// user explicitly configures the environment to do so.
+class VersionUpstreamValidator {
+  VersionUpstreamValidator({
+    required this.version,
+  });
+
+  final FlutterVersion version;
+
+  /// Performs the validation against the tracking remote of the [version].
+  ///
+  /// Throws [VersionCheckError] if the tracking remote is not standard.
+  void run(){
+    // If repositoryUrl is null, exit
+    final String? repositoryUrl = version.repositoryUrl;
+    if (repositoryUrl == null) {
+      throw VersionCheckError(
+        'The tool could not determine the remote upstream which is being '
+        'tracked by the SDK.'
+      );
+    }
+
+    // Strip `.git` suffix before comparing the remotes
+    final String trackingUrl = stripDotGit(repositoryUrl);
+    final String flutterGitUrl = stripDotGit(globals.flutterGit);
+
+    // Exempt the official flutter git SSH remote from this check
+    if (trackingUrl == 'git@github.com:flutter/flutter') {
```

Comment:
rather than hard-coding here, could you also store this in `globals`, next to `flutterGit`?

Reply:
Getting back to this, why does this need to be in `globals`? This and `flutterGit` would be only used in `version.dart`. We could probably remove them from `globals` and instead pass `globals.platform` around.

Reply:
SGTM